### PR TITLE
Add build and host profiles

### PR DESCRIPTION
**Pull request by cschlosser**
Add option to make conan detection silent since it's triggered
three times otherwise when build and host profiles are specified.

### CODE REVIEW

**Comment by a4z** [comment]
what is the status of this PR? 
nothing happens since more than 1 month. is there an other solution for using the new 2 profiles way, or something missing?

**Comment by czoido** [comment]
Hi @christophschlosser thanks a lot for the contribution
Adding this to the milestone of the next release

cc @a4z

**Review by danimtb — APPROVED** [review]

**Review by jgsogo — CHANGES_REQUESTED** [review]
IMHO, I would say that the `--profile:host` (or `--profile`, it is the same) should take the value from any of the existing profile variables: `DEBUG_PROFILE`, `RELEASE_PROFILE`, `RELWITHDEBINFO_PROFILE`, `MINSIZEREL_PROFILE` or `PROFILE` like it is doing right now, and we should **add only the new `BUILD_PROFILE` variable** (not sure if `PROFILE_BUILD` would be a better name).

File: conan.cmake [hunk 1]
```diff
@@ -323,7 +331,7 @@ macro(parse_arguments)
   set(oneValueArgs CONANFILE  ARCH BUILD_TYPE INSTALL_FOLDER CONAN_COMMAND)
   set(multiValueArgs DEBUG_PROFILE RELEASE_PROFILE RELWITHDEBINFO_PROFILE MINSIZEREL_PROFILE
                      PROFILE REQUIRES OPTIONS IMPORTS SETTINGS BUILD ENV GENERATORS PROFILE_AUTO
-                     INSTALL_ARGS CONFIGURATION_TYPES)
+                     INSTALL_ARGS CONFIGURATION_TYPES BUILD_PROFILE HOST_PROFILE)
```
Comment (czoido) [inline conan.cmake:334]:
```suggestion
                     INSTALL_ARGS CONFIGURATION_TYPES PROFILE_BUILD)
```

**Comment by czoido** [comment]
Hi @cschlosser,
I have added some suggestions as I can't push to your branch. Could you please check when possible?

**Review by jgsogo — APPROVED** [review]

**Pull request merged**